src/qt/optionsmodel.cpp

```diff
@@ -19,18 +19,93 @@
 #include <txdb.h>       // for -dbcache defaults
 #include <util/string.h>
 #include <validation.h> // For DEFAULT_SCRIPTCHECK_THREADS
+#include <wallet/wallet.h> // For DEFAULT_SPEND_ZEROCONF_CHANGE
 
 #include <QDebug>
 #include <QLatin1Char>
 #include <QSettings>
 #include <QStringList>
 
+#include <univalue.h>
+
 const char *DEFAULT_GUI_PROXY_HOST = "127.0.0.1";
 
 static const QString GetDefaultProxyAddress();
 
-OptionsModel::OptionsModel(QObject *parent, bool resetSettings) :
-    QAbstractListModel(parent)
+//! Convert bitcoin settings value to integer.
+static int ToInt(const util::SettingsValue& value, int fallback = 0)
+{
+    if (value.isNull()) return fallback;
+    if (value.isBool()) return value.get_bool();
+    if (value.isNum()) return value.get_int();
+    return LocaleIndependentAtoi<int>(value.get_str().c_str());
+}
+
+//! Convert bitcoin settings value to QString.
+static QString ToQString(const util::SettingsValue& value, const QString& fallback = {})
+{
+    if (value.isNull()) return fallback;
+    if (value.isFalse()) return "";
+    return QString::fromStdString(value.get_str());
+}
+
+//! Convert bitcoin setting to QVariant.
+static QVariant ToQVariant(const util::SettingsValue& value, const QVariant& fallback = {})
+{
+    if (value.isNull()) return fallback;
+    if (value.isBool()) return value.get_bool();
+    if (value.isNum()) return value.get_int();
+    return QString::fromStdString(value.get_str());
```

Comment:
Ok, leave it as is then.

I mentioned it because this function has a return type `QVariant` and now it returns `QString`. This, I guess, implicitly calls the `QVariant(const QString &string)` contructor. Where is `QString::fromStdString()` documented? Is the constructor `QVariant(const QString &string)` documented too? I do not see any comments in Qt headers.